homeassistant/components/homekit_controller/switch.py

```diff
@@ -55,3 +47,77 @@ def device_state_attributes(self):
         outlet_in_use = self.service.value(CharacteristicsTypes.OUTLET_IN_USE)
         if outlet_in_use is not None:
             return {OUTLET_IN_USE: outlet_in_use}
+
+
+class HomeKitValve(HomeKitEntity, SwitchDevice):
+    """Represents a valve in an irrigation system."""
+
+    def get_characteristic_types(self):
+        """Define the homekit characteristics the entity cares about."""
+        return [
+            CharacteristicsTypes.ACTIVE,
+            CharacteristicsTypes.IN_USE,
+            CharacteristicsTypes.IS_CONFIGURED,
+            CharacteristicsTypes.REMAINING_DURATION,
+        ]
+
+    async def async_turn_on(self, **kwargs):
+        """Turn the specified valve on."""
+        await self.async_put_characteristics({CharacteristicsTypes.ACTIVE: True})
+
+    async def async_turn_off(self, **kwargs):
+        """Turn the specified valve off."""
+        await self.async_put_characteristics({CharacteristicsTypes.ACTIVE: False})
+
+    @property
+    def icon(self) -> str:
+        """Return the icon."""
+        return "mdi:water"
+
+    @property
+    def is_on(self):
+        """Return true if device is on."""
+        return self.service.value(CharacteristicsTypes.ACTIVE)
+
+    @property
+    def device_state_attributes(self):
+        """Return the optional state attributes."""
+        attrs = {}
+
+        in_use = self.service.value(CharacteristicsTypes.IN_USE)
+        if in_use is not None:
+            attrs[ATTR_IN_USE] = in_use == InUseValues.IN_USE
+
+        is_configured = self.service.value(CharacteristicsTypes.IS_CONFIGURED)
+        if is_configured is not None:
+            attrs[ATTR_IS_CONFIGURED] = is_configured == IsConfiguredValues.CONFIGURED
+
+        remaining = self.service.value(CharacteristicsTypes.REMAINING_DURATION)
+        if remaining is not None:
+            attrs[ATTR_REMAINING_DURATION] = remaining
+
+        return attrs
+
+
+ENTITY_TYPES = {
+    "switch": HomeKitSwitch,
+    "outlet": HomeKitSwitch,
+    "valve": HomeKitValve,
+}
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up Homekit fans."""
```

Comment:
```suggestion
    """Set up Homekit switches."""
```

Reply:
Wow now matter how many times i look i always miss at least one of those! Cheers!